lib/wasi/src/runtime/task_manager/tokio.rs

```diff
@@ -146,6 +146,32 @@ impl VirtualTaskManager for TokioTaskManager {
         Ok(())
     }
 
+    /// See [`VirtualTaskManager::task_wasm_with_trigger`].
+    fn resume_wasm_after_trigger(
+        &self,
+        task: Box<dyn FnOnce(Store, Module) + Send + 'static>,
+        store: Store,
+        module: Module,
+        trigger: Box<
+            dyn FnOnce(Store) -> Pin<Box<dyn Future<Output = TaskResumeAction> + Send + 'static>>
+                + Send
+                + 'static,
+        >,
+    ) -> Result<(), WasiThreadError> {
+        let trigger = trigger(store);
+        let handle = self.0.clone();
+        self.0.spawn(async move {
```

Comment:
Why is resuming bound so tightly to threading?

We shouldn't need to spawn a thread in order to run something resumable.

Reply:
This is required because of how the `postMessage` and friends are implemented in the JS version of the runtime. Suggest when we migrate the runtime to JS that we also refactor this area.

I created an issue for it here:
https://github.com/wasmerio/wasmer/issues/3714